src/serde/visited_nodes.rs

```diff
@@ -0,0 +1,96 @@
+/// This is a simple bitfield used to indicates whether a node has been visited
+/// during a tree search or not. We terminate a search path if we've reached the
+/// node first via a different (shorter) path.
+#[derive(Clone, Default)]
+pub struct VisitedNodes {
+    bits: Vec<usize>,
+}
+
+impl VisitedNodes {
+    const BITS: usize = usize::BITS as usize;
```

Comment:
Why did you decide to not use `BitVec` here?

Reply:
I wasn't very happy with the growth of the dependency tree that came with `BitVec` as well as the complexity. `BitVec` supports a much more general purpose use, whereas `VisitedNodes` is really quite simple. e.g. it only ever grows in size and always by a known number of bits.

Reply:
Might be worth just calling it `BitArray` or something since `VisitedNodes` suggest a specialization that doesn't really exist. A reader gets all they need to know from `BitArray` whereas `VisitedNodes` suggests some extra dark magic.

Reply:
good point. initially this class was only used for one thing